base/math.jl

```diff
@@ -566,7 +566,7 @@ hypot(x::Number...) = _hypot(promote(x...))
 function _hypot(x::NTuple{N,<:Number}) where {N}
 	N == 0 && error("invalid hypot() call without arguments")
 	# speculatively try fast naive approach
-	s = sum(abs2, float.(x))
+	s = sum(abs2.(float.(x)))
```

Comment:
I'm confused that this would be faster than `sum(y -> abs2(float(y)), x)`, which avoids creating a temporary tuple.

Reply:
I think the compiler avoids creating a new tuple and probably does the thing in-place. If it `x` were an `Array`, it would be a different story.